src/Data/ProgrammesDb/EntityRepository/BroadcastRepository.php

```diff
@@ -32,6 +32,34 @@ public function findByVersion(array $dbIds, string $type, ?int $limit, int $offs
         return $qb->getQuery()->getResult(Query::HYDRATE_ARRAY);
     }
 
+    public function findUpcomingByService(int $serviceDbId, string $type, DateTimeImmutable $cutoffTime, ?int $limit, int $offset): array
+    {
+        $qb = $this->createQueryBuilder('broadcast', false)
+            ->addSelect('programmeItem')
+            ->join('broadcast.service', 'service')
+            ->andWhere("IDENTITY(broadcast.service) = :dbId")
+            ->andWhere('broadcast.endAt > :cutoffTime')
+            ->addOrderBy('broadcast.endAt', 'ASC')
```

Comment:
How long does it take to order ALL the future broadcasts?

Reply:
I think @BjornTwachtmann mentioned that filtering where `broadcast.startAt < :cutoffTime` would cut off a load of matches to make the ordering better (as theoretically there should be only one thing active on a service at a time)